qemu/tests/watchdog.py

```diff
@@ -458,3 +458,6 @@ def heartbeat_test():
     else:
         test.error("Oops test %s doesn't exist, have a check please."
                    % test_type)
+    for vm in env.get_all_vms():
+        if vm.is_alive():
+            vm.destroy()
```

Comment:
This step will not run when the test fails. Suggest using a try block to handle it.
```python
    if (test_type in locals()):
        test_running = locals()[test_type]
        try:
            test_running()
        finally:
            vm.destroy()
    else:
        test.error("Oops test %s doesn't exist, have a check please."
                   % test_type)
```

Reply:
Thanks, updated ! Please help to review it again !